### ISSUE TITLE
Nullable parameters in `MemberValueSetValidator`

### ISSUE THREAD

**Issue by bjarnef** [umbraco/Umbraco-CMS#13036]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

10.2.0

### Bug summary

Previous you could customize indexing of content, e.g. to include protected content or customize member index: https://github.com/umbraco/UmbracoDocs/issues/2683

and documented here: https://our.umbraco.com/documentation/Reference/Searching/Examine/indexing/#changing-ivaluesetvalidator

However since it's now .NET6 and nullable enabled by default, this shows warnings, because the parameters can not be null.

https://github.com/umbraco/Umbraco-CMS/blob/v10/contrib/src/Umbraco.Infrastructure/Examine/MemberValueSetValidator.cs#L16-L29

```
using Examine;
using Umbraco.Cms.Core;
using Umbraco.Cms.Core.Composing;
using Umbraco.Cms.Core.DependencyInjection;
using Umbraco.Cms.Core.Scoping;
using Umbraco.Cms.Core.Services;
using Umbraco.Cms.Core.Strings;
using Umbraco.Cms.Infrastructure.Examine;

using Umbraco.Extensions;

namespace MyProject.Core.Search.Index

public class CustomizeIndexComposer : IComposer
{
    public void Compose(IUmbracoBuilder builder)
    {
        // replace the default IUmbracoIndexConfig definition
        //builder.Services.AddUnique<IUmbracoIndexConfig, CustomIndexConfig>();
    }
}

// inherit from the default
public class CustomIndexConfig : UmbracoIndexConfig, IUmbracoIndexConfig
{
    private readonly IShortStringHelper _shortStringHelper;

    public CustomIndexConfig(
        IPublicAccessService publicAccessService,
        IScopeProvider scopeProvider,
        IShortStringHelper shortStringHelper)
        : base(publicAccessService, scopeProvider)
    {
        _shortStringHelper = shortStringHelper;
    }

    // explicit implementation - overrides base class.
    // GetMemberValueSetValidator is used for the MemberIndex.
    IValueSetValidator IUmbracoIndexConfig.GetMemberValueSetValidator()
    {
        // NOTE - by default Umbraco only includes a small subset of fields for the member index for
        // security reasons, however you may want to include a larger set of data in the member index.
        // This example will include all data for a member excluding the built in property values.

        // all built in member properties to exclude (i.e. Password question, etc...)
        var excludeFields = ConventionsHelper.GetStandardPropertyTypeStubs(_shortStringHelper).Keys;

        // include everything except the above
        return new MemberValueSetValidator(null, null, null, excludeFields);
    }
}
```

### Steps to reproduce

Add example  from documentation or above and notice warnings when override e.g. `IUmbracoIndexConfig.GetMemberValueSetValidator()`

![image](https://user-images.githubusercontent.com/2919859/191268502-03a4c590-3bf3-44f6-b975-e7b2680fa4d7.png)

---
_This item has been added to our backlog [AB#23094](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/23094)_

**Comment by bjarnef** [comment]
Also notice the issue with `IScopeProvider` in ctor:
https://github.com/umbraco/Umbraco-CMS/issues/12887

@Zeegaan is it an oversight that the parameters not are nullable? Furthermore I guess we should have an overload of the ctor to accept `Umbraco.Cms.Infrastructure.Scoping.IScopeProvider` instead of the obsolete `Umbraco.Cms.Core.Scoping.IScopeProvider`.

**Comment by Zeegaan** [comment]
@bjarnef Yep seems like its an oversight! 😅 Should probably be fixed for v11 🤔

**Comment by bjarnef** [comment]
I guess it is possible fixing https://github.com/umbraco/Umbraco-CMS/issues/12887 without a breaking change though?

**Comment by Zeegaan** [comment]
Yea should just be a case of obsoleting the old ctor and implementing a new one 👍

**Comment by bjarnef** [comment]
@Zeegaan not sure how we can change the `ScopeProvider` property here?

![image](https://user-images.githubusercontent.com/2919859/191614843-3b4e951b-5cd0-4740-bd73-cb821f1806fb.png)

**Comment by Zeegaan** [comment]
@bjarnef we probably can't :( 
Think the solution is to obsolete the property aswell, and then create a new property called `InfrastructureScopeProvider`, while not the prettiest solution, I can't think of another way sadly

**Comment by bjarnef** [comment]
@Zeegaan unless we could just break it in v11, but I guess it should be marked as obsolete first.

I found another one here https://github.com/umbraco/Umbraco-CMS/blob/v10/contrib/src/Umbraco.Infrastructure/Examine/ContentValueSetBuilder.cs#L33

In this case we can add an overload and I guess it should call the new one, but not sure if `Umbraco.Cms.Core.Scoping.IScopeProvider` can be casted to `Umbraco.Cms.Infrastructure.Scoping.IScopeProvider` or it is fine to use `StaticServiceProvider.Instance.GetRequiredService<IScopeProvider>()` although the parameter isn't used (except where implemented).

![image](https://user-images.githubusercontent.com/2919859/191674791-60de7368-c9c7-4992-8910-647fe3bf4261.png)

**Comment by Zeegaan** [comment]
@bjarnef Don't think we could break it sadly :(
Yea I think its fine to use the `StaticServiceProvider.Instance.GetRequiredService<IScopeProvider>()` 👍

**Comment by nikolajlauridsen** [comment]
This has been fixed in #13185, thanks for reporting 😄
